### DISCUSSION TITLE
Expression produces a union type that is too complex to represent.ts(2590)

### DISCUSSION THREAD

**Discussion by mohammad-fahad** [tailwindlabs/headlessui#1742] [Help] [answered]
I am using NX MonoRepo with 

- Next.js: "12.1.6", 
- React: "18.2.0"
- typeScript:  "4.7.4", and 
- "@headlessui/react": "^1.6.6", 

I tried the solution of turning reactStrictMode: false but still, I can't use it. I am trying to use a Dialog but it says "Expression produces a union type that is too complex to represent.ts(2590)". 
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/63115051/183071565-db5e1894-51f3-4b10-842d-7e5bc5b0f314.png">

**Comment by mohammad-fahad** [comment, marked as answer]
@RobinMalfait 
@adamwathan 
Please help me!! It is URGENT!

**Reply by adamwathan** [reply]
This just looks like an editor lint error, according to the console at the bottom the project is compiling no? Seems like a TypeScript version thing.

Also chill.

**Reply by mohammad-fahad** [reply]
Thanks, @adamwathan !!! 
It is woking!!

**Reply by MuttakinHasib** [reply]
@adamwathan  , getting build error in production. How can I fix this.

![image](https://user-images.githubusercontent.com/44552983/184456699-fd20575b-fe25-4551-a050-dc6612b47aa4.png)
